### ISSUE TITLE
Set legend title font properties

### ISSUE THREAD

**Issue by smartlixx** [matplotlib/matplotlib#19259]
### Problem
I recently want to set the legend title font properties (say, use Chinese font for Chinese title) but found the `prop` argument only applies to legend entries, not legend title. For example:

```python
     import matplotlib.pyplot as plt
     import matplotlib as mpl
     myfont1 = mpl.font_manager.FontProperties(fname='/System/Library/Fonts/Supplemental/Arial Italic.ttf')     

     fig, [ax1, ax2] = plt.subplots(1, 2, figsize=(8, 3))
     
     ax1.scatter(1, 1, 40, label='point')
     ax1.legend(title='location')
     
     ax2.scatter(1, 1, 40, label='point')
     ax2.legend(title='location', prop=myfont1)
     plt.show()
```

This will produces the following plot, whch is not what I want. 

![Legend_title_1](https://user-images.githubusercontent.com/16891009/103994368-dfd19080-51d1-11eb-943b-4022d1892658.png)

One alternative is to use `set_title` to set title font properties, for example:

```python
     import matplotlib.pyplot as plt
     import matplotlib as mpl
     myfont1 = mpl.font_manager.FontProperties(fname='/System/Library/Fonts/Supplemental/Arial Italic.ttf')     
     myfont2 = mpl.font_manager.FontProperties(fname='/System/Library/Fonts/Supplemental/Arial Bold.ttf')

     fig, [ax1, ax2] = plt.subplots(1, 2, figsize=(8, 3))
     
     ax1.scatter(1, 1, 40, label='point')
     ax1.legend(title='location')
     
     ax2.scatter(1, 1, 40, label='point')
     l = ax2.legend(title='location', prop=myfont1)
     l.set_title(title='location', prop=myfont2)
     plt.show()
```

This code produces what I want, but not in a consistent way as we prescribe the font properties of legend entry, especially when matplotlib is called by other pacages (say, by geopandas's plot function, where a `legend_kws` argument will be passed to matplotlib to produce legend. I found this problem when I used geopandas to plot a choropleth map).

![Legend_title_2](https://user-images.githubusercontent.com/16891009/103994991-c0873300-51d2-11eb-8c97-df4a6f92b211.png)

### Proposed Solution

It would be great to add an additional argumnt `title_prop` to `legend()` to control the title font properties. Currently, matplotlib seems to only care about title font size: ([legend.py](https://github.com/matplotlib/matplotlib/blob/master/lib/matplotlib/legend.py), line 509-514)

```python
        # figure out title fontsize:
        if title_fontsize is None:
            title_fontsize = mpl.rcParams['legend.title_fontsize']
        tprop = FontProperties(size=title_fontsize)
        self.set_title(title, prop=tprop)
        self._draggable = None
```  

### Additional context and prior art
The title font properties are important when we have a lgend title not in English, and we don't want to rely on the rcParams to change the global font properties. If the proposed addition is acceptable, I may try to make a PR to address this issue.

**Comment by jklymak** [comment]
I think thats possible - we discussed it when we put in `title_fontsize`, and the issue was simply that it adds to an already voluminous set of kwargs.  There is also an issue to `title_fontprops` conflicting w/ `title_size`, but we do that elsewhere as well, I think, so you can just copy that.

**Comment by QuLogic** [comment]
Closed by #19304.
